File: src/main/scala/org/nygenome/als/graphdb/value/NeurobankSubjectProperty.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord

case class NeurobankSubjectProperty ( subjectId:Int,
                                      subjectGuid:String,
                                      genomicDataFlag:Boolean,
                                      eventCategory:String,
                                      eventPropertyCode:String,
                                      eventPropertyName:String,
                                      eventPropertyValue:String
                                    ) {
  val id:String = eventCategory +":" +eventPropertyCode

}

object NeurobankSubjectProperty extends ValueTrait {
  def parseCSVRecord(record:CSVRecord):NeurobankSubjectProperty = {
    new  NeurobankSubjectProperty(record.get("subject_id").toInt,
    record.get("subject_guid"),
      booleanValueFromInt(record.get("genomic_data_flag").toInt),
      record.get("category"),
      record.get("property_code"),
      record.get("property_name"),
      record.get("property_value"))

  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/NeurobankSubjectProperty.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class NeurobankSubjectProperty(
        val subjectId: String,
        val subjectGuid: String,
        val genomicDataFlag: Boolean,
        val eventCategory: String,
        val eventPropertyCode: String,
        val eventPropertyName: String,
        val eventPropertyValue: String
) {
    val id = "$subjectGuid:$eventCategory:$eventPropertyCode"
    val subjectIdPair = Pair(subjectId, subjectGuid)

    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): NeurobankSubjectProperty =
                NeurobankSubjectProperty(record.get("subject_id"),
                        record.get("subject_guid"),
                        booleanFromInt(record.get("genomic_data_flag").toInt()),
                        record.get("category"),
                        record.get("property_code"),
                        record.get("property_name"),
                        record.get("property_value"))
    }
}